nixos/pam: allow changing password using sssd

s1341 opened the pull request:
###### Description of changes
Modify the pam configuration for sssd to `try_first_pass` instead of `use_authtok`, as the latter swallows password change prompts.

Apologies for the previous pull request. I mistakenly targeted the wrong branch!

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
lib/systems: add mips64[el] entries to qemuArch

This commit adds `mips64el` to the `qemuArch` table.

takov751 committed:
obs-studio-plugins.advanced-scene-switcher: 1.20.5 -> 1.21.1

s1341 committed:
nixos/pam: allow changing password using sssd

s1341 commented:
@ofborg build

mweinelt commented:
@ofborg eval

fabaff committed:
python310Packages.archinfo: 9.2.48 -> 9.2.49

fabaff committed:
python310Packages.ailment: 9.2.48 -> 9.2.49

fabaff committed:
python310Packages.pyvex: 9.2.48 -> 9.2.49

fabaff committed:
python310Packages.claripy: 9.2.48 -> 9.2.49

fabaff committed:
python310Packages.cle: 9.2.48 -> 9.2.49

fabaff committed:
python310Packages.angr: 9.2.48 -> 9.2.49

RaitoBezarius commented:
It would be ideal if we had NixOS tests for this, I am not knowledgeable enough about PAM to bring any relevant input.

r-ryantm committed:
python310Packages.mesa: 1.2.0 -> 1.2.1

jtojnar committed:
wireshark: do not install redundant files

The icon we were installing is an old one that should not be used these days: https://gitlab.com/wireshark/wireshark/-/merge_requests/10507
And the CMake build script already installs the desktop file and icons in a more proper location since 2.4.0: https://gitlab.com/wireshark/wireshark/-/commit/c27404655650a3648f576663d07394105fd5ef30

Commit:
release-cross.nix: fix the fix in #188339

The `bootstrapTools.bootstrapTools` attribute, which *unpacks* the
bootstrapFiles, lies about its `system`:

  https://github.com/NixOS/nixpkgs/blob/78963777a40a63333520c5c8f4567b4059b68c4d/pkgs/stdenv/linux/make-bootstrap-tools.nix#L236

Unfortunately because of this, when building `bootstrapTools.*`,
Hydra will try to execute the cross-compiled bootstrapFiles'
`busybox` on the wrong architecture, and fail.

I originally submitted a fix for this, by filtering out the
`.bootstrapTools` attribute, in
https://github.com/NixOS/nixpkgs/pull/188339

Unfortunately my fix filtered at the wrong level of the attrset
hierarchy, and neglected to filter the `test` attribute as well.
This PR corrects those mistakes

Commit:
cc-wrapper: when merging gcc32 and gcc64, merge libgcc as well

Our gcc_multi and glibc_multi expressions merge together a
32-bit-targeted and 64-bit-targeted gcc.  However they do not thread
through the passthru.libgcc from these merged gccs.

This commit corrects that.

It also extends passthru.libgcc to allow a *list* rather than just a
single outpath.

Resolves part of #221891 (at least getting it back to the error
message it gave before).

risicle committed:
libressl_3_5: remove

risicle committed:
libressl: 3.6.2 -> 3.7.2

mweinelt committed:
python310Packages.defusedcsv: init at 2.0.0

mweinelt committed:
python310Packages.django-i18nfield: init at 1.9.4

mweinelt committed:
python310Packages.drf-ujson2: init at 1.7.2

mweinelt committed:
python310Packages.paypalhttp: init at 1.0.0

mweinelt committed:
python310Packages.paypal-checkout-serversdk: init at 1.0.

mweinelt committed:
python310Packages.django-mysql: init at 4.9.0

r-ryantm committed:
slweb: 0.5 -> 0.5.4

r-ryantm committed:
pgbouncer: 1.18.0 -> 1.19.0

r-ryantm committed:
praat: 6.3.09 -> 6.3.10

fabaff committed:
python311Packages.cle: use binaries 9.2.49

vbgl committed:
ocamlPackages.lambdasoup: 0.7.3 → 1.0.0

Niols committed:
ocamlPackages.ppx_deriving_qcheck: init at 0.20

peterhoeg committed:
ppp: specify statedir for .pid file

peterhoeg committed:
openfortivpn: 1.20.2 -> 1.20.3

figsoda committed:
cargo-raze: fix build on darwin

mweinelt committed:
python310Packages.python-u2flib-server: init at 5.0.1

mweinelt committed:
python310packages.pyuca: init at 1.2

mweinelt committed:
python310Packages.slimit: init at unstable-2018-08-08

malob committed:
signalbackup-tools: 20230429 -> 20230508-1

Diff: https://github.com/bepaald/signalbackup-tools/compare/20230429...20230508-1

mweinelt committed:
python310Packages.static3: init at 0.7.0

wegank committed:
gnatcoll-core: fix build on x86_64-darwin

wegank committed:
gnatcoll-python3: fix build on x86_64-darwin

fabaff committed:
skjold: 0.4.1 -> 0.6.1

Changelog: https://github.com/twu/skjold/releases/tag/v0.6.1

fabaff committed:
inql: fix invalid version specifier

wegank committed:
gdal: fix build on x86_64-darwin

fabaff committed:
python310Packages.avion: fix specifier

SuperSandro2000 committed:
nixos/fzf: fix evaluation if ohMyZsh is enabled

Aleksanaa committed:
gabutdm: init at 2.1.5

mweinelt committed:
python310Packages.vat-moss: init at 0.11.0

And apply a patch that updates the EU API endpoint to use HTTPS.

r-ryantm committed:
sarasa-gothic: 0.40.6 -> 0.40.7

vbgl committed:
ocaml-top: use Dune 3

vbgl committed:
ocamlPackages.ocp-index: use Dune 3

vbgl committed:
ocamlPackages.ocp-indent: use Dune 3

r-ryantm committed:
flyctl: 0.0.558 -> 0.0.559

r-ryantm committed:
hcxdumptool: 6.1.4 -> 6.3.0

r-ryantm committed:
go-containerregistry: 0.14.0 -> 0.15.1

r-ryantm committed:
python310Packages.pyswitchbee: 1.7.26 -> 1.8.0

r-ryantm committed:
python310Packages.rq: 1.13.0 -> 1.14.1

r-ryantm committed:
jetbrains-toolbox: 1.28.0.15158 -> 1.28.1.15219

fabaff committed:
Merge pull request #230658 from fabaff/angr-bump

python310Packages.angr: 9.2.48 -> 9.2.49

fabaff committed:
Merge pull request #230742 from fabaff/skjold-bump

skjold: 0.4.1 -> 0.6.1

fabaff committed:
Merge pull request #230752 from fabaff/inql-fix

inql: fix invalid version specifier

fabaff committed:
Merge pull request #230763 from fabaff/avion-fix

python310Packages.avion: fix specifier

fpletz committed:
Merge pull request #230562 from risicle/ris-libressl-3.7

libressl: 3.6.2 -> 3.7.2

fpletz committed:
Merge pull request #230397 from NixOS/wip/jtojnar/wireshark-cleanup

Commit:
Merge pull request #228013 from amjoseph-nixpkgs/pr/qemuArch/mips

lib/systems: add mips64[el] entries to qemuArch

Commit:
wrapClangMulti: set gcc_multi_sysroot.langCC

The phoney gcc that we construct for multilib was missing the
`langCC` attribute, which `cc-wrapper` needs in order to decide
whether or not to add gcc's `libstdc++` headers as an `-isystem` for
`clang`.

Commit:
bazarr: 1.2.0 -> 1.2.1

Commit:
wrapClangMulti: link include/c++ into gcc_multi_sysroot

The phoney gcc that we construct for multilib was missing the
`$out/include/c++` directory which `cc-wrapper` needs to pass as an
`-isystem` to `clang`.

Closes #221891

r-ryantm committed:
ocenaudio: 3.11.24 -> 3.11.25

r-ryantm committed:
python310Packages.weasyprint: 58.0 -> 58.1

wegank committed:
python310Packages.jaxlib: mark as broken on darwin

wegank committed:
Merge pull request #230781 from r-ryantm/auto-update/sarasa-gothic

sarasa-gothic: 0.40.6 -> 0.40.7

r-ryantm committed:
python310Packages.yolink-api: 0.2.8 -> 0.2.9

wegank committed:
Merge pull request #230759 from Aleksanaa/gabutdm

gabutdm: init at 2.1.5

Commit:
Merge pull request #230470 from amjoseph-nixpkgs/fix/221891

cc-wrapper: when merging gcc32 and gcc64, merge libgcc as well

wegank committed:
Merge pull request #230823 from NULLx76/bazarr-1.2.1

bazarr: 1.2.0 -> 1.2.1

wegank committed:
Merge pull request #230455 from amjoseph-nixpkgs/zhf/bootstrapTools/dont-unpack-foreign

release-cross.nix: fix the fix in #188339

lheckemann committed:
Merge pull request #228065 from takov751/master

obs-studio-plugins.advanced-scene-switcher: 1.20.5 -> 1.21.1

wegank committed:
python311Packages.hepmc3: fix build

madjar committed:
Merge pull request #230687 from peterhoeg/u/openfortivpn_1_20_3

openfortivpn: 1.20.2 -> 1.20.3

wegank committed:
Merge pull request #230814 from wegank/jaxlib-darwin

python310Packages.jaxlib: mark as broken on darwin

ulrikstrid committed:
Merge pull request #230755 from vbgl/ocaml-lambdasoup-1.0.0

ocamlPackages.lambdasoup: 0.7.3 → 1.0.0

wegank committed:
Merge pull request #230815 from r-ryantm/auto-update/python310Packages.pyswitchbee

python310Packages.pyswitchbee: 1.7.26 -> 1.8.0

wegank committed:
Merge pull request #230831 from wegank/root-python

python311Packages.hepmc3: fix build

wegank committed:
Merge pull request #230754 from wegank/tiledb-darwin

gdal: fix build on x86_64-darwin

wegank committed:
Merge pull request #230710 from wegank/gnatcoll-core-hack

gnatcoll-core: fix build on x86_64-darwin

wegank committed:
Merge pull request #230810 from r-ryantm/auto-update/flyctl

flyctl: 0.0.558 -> 0.0.559

wegank committed:
Merge pull request #230811 from r-ryantm/auto-update/hcxdumptool

hcxdumptool: 6.1.4 -> 6.3.0

wegank committed:
Merge pull request #230812 from r-ryantm/auto-update/go-containerregistry

go-containerregistry: 0.14.0 -> 0.15.1

wegank committed:
Merge pull request #230816 from r-ryantm/auto-update/python310Packages.rq

python310Packages.rq: 1.13.0 -> 1.14.1

wegank committed:
Merge pull request #230824 from r-ryantm/auto-update/ocenaudio

ocenaudio: 3.11.24 -> 3.11.25

wegank committed:
Merge pull request #230825 from r-ryantm/auto-update/python310Packages.weasyprint

python310Packages.weasyprint: 58.0 -> 58.1

wegank committed:
Merge pull request #230826 from r-ryantm/auto-update/python310Packages.yolink-api

python310Packages.yolink-api: 0.2.8 -> 0.2.9

wegank committed:
Merge pull request #230817 from r-ryantm/auto-update/jetbrains-toolbox

jetbrains-toolbox: 1.28.0.15158 -> 1.28.1.15219

wegank committed:
Merge pull request #230590 from r-ryantm/auto-update/slweb

slweb: 0.5 -> 0.5.4

matthiasbeyer committed:
peertube: Remove matthiasbeyer from maintainers

I do not feel qualified of maintaining this.

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

wegank committed:
Merge pull request #230591 from r-ryantm/auto-update/pgbouncer

pgbouncer: 1.18.0 -> 1.19.0

wegank committed:
Merge pull request #230637 from r-ryantm/auto-update/praat

praat: 6.3.09 -> 6.3.10

wegank committed:
Merge pull request #230736 from malob/update-signalbackup-tools

signalbackup-tools: 20230429 -> 20230508-1

wegank committed:
Merge pull request #230286 from r-ryantm/auto-update/python310Packages.mesa

python310Packages.mesa: 1.2.0 -> 1.2.1

SuperSandro2000 committed:
Merge pull request #230756 from NixOS/fzf-fix

mweinelt committed:
Merge pull request #230707 from mweinelt/varios-python-deps

Various python packages

wegank committed:
Merge pull request #230809 from vbgl/ocaml-ocp-indent-dune3

ocamlPackages.ocp-indent: fix for Dune 3

alyssais committed:
pkgsStatic.libhwy: fix build (#230761)

This was changed in 7328b78fb58, but I don't understand why — the only
difference should be in static builds, which worked before that change
and didn't work after.

Fixes: 7328b78fb58 ("libhwy: 0.15.0 -> 1.0.2")

wegank committed:
Merge pull request #230728 from figsoda/cargo-raze

cargo-raze: fix build on darwin

bobby285271 committed:
Merge pull request #230840 from matthiasbeyer/remove-me-peertube-maintainer

peertube: Remove matthiasbeyer from maintainers

s1341 committed:
pam: remove unused try_first_pass

s1341 committed:
nixos/test/sssd-ldap: make tests actually check login, password change

s1341 commented:
Damnit... I rebased master by accident! Apologies to all... I'll reopen the PR.
